src/Microsoft.AspNetCore.Razor/CodeGenerators/Visitors/CSharpTagHelperFieldDeclarationVisitor.cs

```diff
@@ -168,6 +188,23 @@ private void PreAllocateUnboundTagHelperAttributes(TagHelperChunk chunk)
             }
         }
 
+        private static bool IsLiteralAttributeValue(Chunk attributeValueChunk)
```

Comment:
CSharpTagHelperCodeRender is checking for IsDynamicValue. Here we might need to check for IsLiteralChunk. So I added this method